frontend/web/components/modals/base/Modal.tsx

```diff
@@ -0,0 +1,112 @@
+import {
+  Modal as _Modal,
+  ModalBody as _ModalBody,
+  ModalFooter as _ModalFooter,
+  ModalHeader as _ModalHeader,
+  ModalProps,
+} from 'reactstrap'
+import { JSXElementConstructor, ReactNode, useCallback, useState } from 'react'
+import { render, unmountComponentAtNode } from 'react-dom'
+import Confirm from './ModalConfirm'
+import ModalDefault, { interceptClose } from './ModalDefault'
+import Alert from './ModalAlert'
+import { getStore } from 'common/store'
+import { Provider } from 'react-redux'
+
+export const ModalHeader = _ModalHeader
+export const ModalFooter = _ModalFooter
+export const Modal = _Modal
+export const ModalBody = _ModalBody
+
+const withModal = (
+  WrappedComponent: JSXElementConstructor<any>,
+  closePointer = 'closeModal',
+) => {
+  return (props: ModalProps) => {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const [isOpen, setIsOpen] = useState(true)
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const toggle = useCallback(() => {
+      if (interceptClose) {
+        interceptClose().then(() => {
+          setIsOpen(false)
+        })
+      } else {
+        setIsOpen(false)
+      }
+    }, [setIsOpen])
+    // @ts-ignore
+    global[closePointer] = toggle
+
+    return (
+      <Provider store={getStore()}>
+        <WrappedComponent toggle={toggle} {...props} isOpen={isOpen} />
+      </Provider>
+    )
+  }
+}
+
+const _Confirm = withModal(Confirm)
+const _ModalDefault2 = withModal(ModalDefault, 'closeModal2')
+const _ModalDefault = withModal(ModalDefault)
+
+export const openConfirm = (global.openConfirm = (
+  title: string,
+  body: ReactNode,
+  onYes?: () => void,
+  onNo?: () => void,
+) => {
+  document.getElementById('confirm') &&
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    unmountComponentAtNode(document.getElementById('confirm')!)
+  render(
+    <_Confirm isOpen onNo={onNo} onYes={onYes} title={title}>
+      {body}
+    </_Confirm>,
+    document.getElementById('confirm'),
+  )
+})
+
+export const openModal = (global.openModal = (
+  title: string,
+  body: ReactNode,
+  className?: string,
+  onClose?: () => void,
+) => {
+  document.getElementById('modal') &&
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    unmountComponentAtNode(document.getElementById('modal')!)
+  render(
+    <_ModalDefault
+      isOpen
+      className={className}
+      onClosed={onClose}
+      title={title}
+    >
+      {body}
+    </_ModalDefault>,
+    document.getElementById('modal'),
+  )
+})
+
+export const openModal2 = (global.openModal2 = (
```

Comment:
Can we add any context about what the difference is between `openModal` and `openModal2` ? I assume perhaps that the idea is that `openModal` will be removed eventually in favour of `openModal2` but I don't want this to drag on and the context to get lost.

Reply:
There are a few areas where there are modals on modals. This already existed prior to the PR.

Reply:
Added comment